guides/source/active_support_structured_events.md

```diff
@@ -0,0 +1,722 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON <https://guides.rubyonrails.org>.**
+
+Active Support Structured Events
+================================
+
+Active Support provides a unified API for reporting *structured events* inside your application. Structured events capture named facts about what happened, when, and from where. They form the foundation for Rails' observability story and can be exported to any external provider.
+
+This guide explains the concepts, design, and usage patterns behind structured events, and how Rails integrates them with framework instrumentation.
+
+After reading this guide, you will understand:
+
+* What structured events are
+* When to use them
+* How Rails reports events
+* How subscribers consume and export events
+* How tags and context work
+* How structured subscribers connect Notifications -> Events
+* Where to find the authoritative API definitions
+* The structured events emitted by Rails (framework hooks)
+
+-------------------------------------------------------------
+
+Introduction to Structured Events
+---------------------------------
+
+Structured events are machine-friendly, semantically consistent records describing something that occurred in your application. They are designed for production observability, fast search, analytics platforms, and developer-facing diagnostic tools.
+
+Each event contains:
+
+* a **name**
+* a **payload** (hash or event object)
+* **timestamp** (nanosecond precision)
+* **source location**
+* **tags**
+* **context**
+
+Rails automatically attaches timestamps, source location, and context so developers only need to provide a name and payload.
+
+### Why Structured Events?
+
+Rails already emits unstructured logs. They are ideal for development but become difficult and expensive to use in production. Some of the challenges of logs are as follows:
+
+* logs from many threads and processes interleave
+* there is no consistent schema
+* specialized knowledge is needed to query the logs
+* indexing costs grow with volume
+* onboarding requires an understanding of bespoke log formats
+
+Structured events fix these problems as they:
+
+* have a consistent shape
+* have semantic naming
+* have predictable payload keys
+* are cheap to index in structured stores
+* are easy to query
+* are compatible with any Obeservability backend
```

Comment:
```suggestion
* are compatible with any Observability backend
```

Although maybe I'd say "integrate easily with Observability backends" instead? ie. in the case of event objects, subscribers may need to serialize them into a shape that ends up being compatible with the observability backend being used.